src/main/java/mate/academy/spring/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,50 @@
+package mate.academy.spring.security;
+
+import java.util.ArrayList;
+import java.util.List;
+import mate.academy.spring.dao.UserDao;
+import mate.academy.spring.entity.Role;
+import mate.academy.spring.entity.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    @Autowired
+    private UserDao userDao;
+
+    @Override
+    public UserDetails loadUserByUsername(String username)
+            throws UsernameNotFoundException {
+
+        User user = userDao.findByUsername(username);
+        if (user == null) {
```

Comment:
@RomanMinevich Can we return optional _ findByUsername_?

Reply:
Done